### PR TITLE
Add coverage ci integration

### PR DESCRIPTION
**Pull request by obukhov**

### CODE REVIEW

File: fleet/client.go [hunk 1]
```diff
@@ -1,17 +1,61 @@
...
+
 	return &Client{
-		URL: url,
+		baseUrl:       baseUrl,
+		err:           err,
+		requestSender: &httpRequestSender{http.Client{}},
+	}
+}
+
+func NewValidClient(rawUrl string) (*Client, error) {
```
Comment (spesnova) [inline fleet/client.go:35]: I think you don't need to define this function to validate client initialization error.
You can make `NewClient()` returns `err` simply and just check it like this.

``` go
# fleet/client.go
func NewClient(rawUrl string) *Client, err {
    ...
    ...
    return &Client{
        baseUrl: baseUrl,
        requestSender: &httpRequestSender{http.Client{}},
}
```

``` go
# fleet/client_test.go
func TestNewClientValid(t *testing.T) {
    client, err := NewClient("http://valid-url.com:5000/fleet")

         if nil != err {
            t.Error("Error supposed to be nil")
         }
         ...
         ...
}
```

How do you think?

**Comment by spesnova** [comment]
Let me refactor this project to enable to test 😃

File: fleet/client.go (hunk 1 above)
Comment (obukhov) [reply fleet/client.go:35]: Yeah I was thinking about it, but this will break backwards compatibility unfortunately. That's why I introduced new function.

**Comment by obukhov** [comment]
Actually I've sent this PR accidently I'm currently working on covering test in my fork. 
Take a look: https://github.com/obukhov/go-fleet/pull/4

I achieved 54% of coverage, but want to reach at least 80% and then will send PR to your repo.

**Comment by obukhov** [comment]
I'd close it for now not to create a mess :)

**Pull request closed without merging**